torch/csrc/jit/codegen/cuda/root_domain_map.h

```diff
@@ -454,6 +457,27 @@ class TORCH_CUDA_CU_API ComputeAtRootDomainMapBuilder
   bool map_through_reduction_ = false;
 };
 
+//! Maps root domains of an entire fusion. Does not map broadcast
+//! domains with non-broadcast domains.
+class TORCH_CUDA_CU_API ExactRootDomainMap : public RootDomainMap {
+ public:
+  ExactRootDomainMap(Fusion* fusion);
+
+  bool areMapped(const IterDomain* id_a, const IterDomain* id_b) const;
+
+  std::string toString() const;
+
+ protected:
+  std::unordered_map<IterDomain*, IterDomain*> map(
```

Comment:
What's the use case for this function? Maybe I missed it but don't see it called anywhere.

Reply:
The parent class, `RootDomainMap`, has it as a pure virtual function, which is used by `RootDoaminMap::mapProducerToConsumer`, etc.